Nodsoft.WowsReplaysUnpack.Generators/SerializableEntity/SerializableEntityGenerator.cs

```diff
@@ -0,0 +1,164 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Nodsoft.WowsReplaysUnpack.Generators.SerializableEntity;
+
+[Generator]
+public class SerializableEntityGenerator : IIncrementalGenerator
+{
+	private const string SerializableEntityAttribute =
+		"Nodsoft.WowsReplaysUnpack.Generators.SerializableEntityAttribute";
+
+	private const string DataMemberAttribute = "System.Runtime.Serialization.DataMemberAttribute";
+	public void Initialize(IncrementalGeneratorInitializationContext context)
+	{
+		context.RegisterPostInitializationOutput(ctx => ctx.AddSource(
+			"SerializableEntityAttribute.g.cs",
+			SourceText.From(SourceGenerationHelper.SerializableEntityAttribute, Encoding.UTF8)));
+
+		IncrementalValuesProvider<EntityToGenerate?> provider = context.SyntaxProvider
+			.ForAttributeWithMetadataName(SerializableEntityAttribute,
+				predicate: (node, _) => node is ClassDeclarationSyntax,
```

Comment:
Could be simplified to just returning true since the current condition is already guaranteed by the compiler through the attribute targets of the attribute